File: JavaScript/7kyu/C.Wars.js
function initials(n){
  let arr = n.split` `
  let final = ``
  
  for(let i = 0; i < arr.length - 1; ++i){
    final += arr[i][0].toUpperCase() + `.`
  }
  
  return final + (x=>x[0].toUpperCase() + x.slice(1).toLowerCase())(arr[arr.length-1])
}

File: Ruby/7kyu/C.Wars.rb
def initials(name)
  arr = name.split(" ")
  i = 0
  result = ""  
  
  while i < arr.length() - 1 do
    puts arr[i]
    result = result + arr[i][0].upcase + "."
    i += 1
  end
  
  final = arr[arr.length-1]
  
  return result + final[0].upcase + final.slice(1,final.length).downcase
end
